src/data_info/services/stat.clj

```diff
@@ -27,6 +29,20 @@
   [stat-map]
   (= (:permission stat-map) :own))
 
+(defn- needs-key?
+  [requested-keys needed-key?]
+  (or (contains? requested-keys needed-key?)
+    (case needed-key?
+      ; :permission is needed by anything which uses `owns?`
+      :permission (contains? requested-keys :share-count)
+      ; :type is needed by anything which uses `is-dir?`
+      :type       (not (empty? (cset/intersection #{:infoType :content-type :file-count :dir-count} requested-keys)))
+      false)))
```

Comment:
This one I did deliberately -- as @psarando pointed out to me at some point when I made a similar suggestion, it's a little silly to force memorizing "oh, `seq` means non-empty" when you're just testing for non-emptiness. This is explicit about what it means!